### PR TITLE
Redact visible passwords in yb-voyager.log

### LINKED ISSUE #386: Do not log clear text passwords

### PR DESCRIPTION
**Pull request by rahulb-yb**
[Fixes #386]
This patch ensures to redact all (possible occurrences of) visible passwords within yb-voyager.log across all logging levels.
Note that there are two redact packages being used, one for URLs, and one for structs.

### CODE REVIEW

File: yb-voyager/src/srcdb/pg_dump_export_data.go [hunk 1]
```diff
@@ -39,8 +39,10 @@ func pgdumpExportDataOffline(ctx context.Context, source *Source, connectionUri
 	// Using pgdump for exporting data in directory format.
 	cmd := fmt.Sprintf(`pg_dump "%s" --no-blobs --data-only --no-owner --compress=0 %s -Fd --file %s --jobs %d --no-privileges --no-tablespaces`,
 		connectionUri, tableListPatterns, dataDirPath, source.NumConnections)
-
-	log.Infof("Running command: %s", cmd)
+	redactedUri := utils.GetRedactedURLs([]string{connectionUri})[0]
+	redactedCmd := fmt.Sprintf(`pg_dump "%s" --no-blobs --data-only --no-owner --compress=0 %s -Fd --file %s --jobs %d --no-privileges --no-tablespaces`,
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/pg_dump_export_data.go:43]: Each time you change the main `cmd`, you will have to change the `redactedCmd`. What if we do something like:
```
pgDumpArgs := fmt.Sprintf("--no-blobs --data-only --no-owner --compress=0 %s -Fd --file %s --jobs %d --no-privileges --no-tablespaces", ...)
cmd := fmt.Sprintf(`pg_dump "%s" %s`, connectionUri, pgDumpArgs)
redactedCmd := fmt.Sprintf(`pg_dump "%s" %s`, redactedUri, pgDumpArgs)
```

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go [hunk 2]
```diff
@@ -18,7 +18,10 @@ func pgdumpExtractSchema(schemaList string, connectionUri string, exportDir stri
 
 	cmd := fmt.Sprintf(`pg_dump "%s" --schema-only --schema "%s" --no-owner -f %s --no-privileges --no-tablespaces`,
 		connectionUri, schemaList, filepath.Join(exportDir, "temp", "schema.sql"))
-	log.Infof("Running command: %s", cmd)
+	redactedUri := utils.GetRedactedURLs([]string{connectionUri})[0]
+	redactedCmd := fmt.Sprintf(`pg_dump "%s" --schema-only --schema "%s" --no-owner -f %s --no-privileges --no-tablespaces`,
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/pg_dump_extract_schema.go:22]: Same as above.

File: yb-voyager/src/tgtdb/target.go [hunk 3]
```diff
@@ -95,3 +98,10 @@ func generateSSLQueryStringIfNotExists(t *Target) string {
 	}
 	return SSLQueryString
 }
+
+func GetRedactedTarget(t *Target) *Target {
+	redactedTarget := *t
+	redactedTarget.Uri = utils.GetRedactedURLs([]string{t.Uri})[0]
+	redact.Redact([]string{"Password"}, &redactedTarget)
```
Comment (amit-yb) [inline yb-voyager/src/tgtdb/target.go:105]: How about setting the Password field directly ;-) 

    redactedTarget.Password = "XXX"

Then you can remove the dependency on the `redact` module.

File: yb-voyager/cmd/importData.go [hunk 4]
```diff
@@ -131,7 +131,8 @@ func getYBServers() []*tgtdb.Target {
 			}
 
 			clone.Uri = getCloneConnectionUri(clone)
-			log.Infof("using yb server for import data: %+v", clone)
+			redactedClone := tgtdb.GetRedactedTarget(clone)
+			log.Infof("using yb server for import data: %+v", redactedClone)
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:135]: You can combine these two statements:

    log.Infof("...", tgtdb.GetRedactedTarget(clone))

File: yb-voyager/cmd/importData.go [hunk 5]
```diff
@@ -211,7 +212,7 @@ func testYbServers(targets []*tgtdb.Target) {
 		utils.ErrExit("no yb servers available/given for data import")
 	}
 	for _, target := range targets {
-		log.Infof("testing server: %s\n", spew.Sdump(target))
+		log.Debugf("testing server: %s\n", spew.Sdump(tgtdb.GetRedactedTarget(target)))
```
[the remark this answers is not in the document]
Comment (amit-yb) [reply yb-voyager/cmd/importData.go:215]: Do we log the `Source` struct somewhere?

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 2 above)
Comment (amit-yb) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:22]: Do we log ora2pg command line in a similar manner?

File: yb-voyager/cmd/importData.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (rahulb-yb) [reply yb-voyager/cmd/importData.go:215]: Never in its entirety, no- its always relevant parts which usually comprises of a portion of the source struct

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 2 above)
Comment (rahulb-yb) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:22]: For ora2pg, all the config options are edited into the ora2pg.conf template file, the shell command itself contains no sensitive information

File: yb-voyager/go.mod [hunk 6]
```diff
@@ -48,6 +48,7 @@ require (
 	github.com/spf13/jwalterweatherman v1.1.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/subosito/gotenv v1.2.0 // indirect
+	github.com/ucarion/redact v0.0.0-20190527210305-5ee85e474909 // indirect
```
Comment (amit-yb) [inline yb-voyager/go.mod:51]: Now, do you need this dependency?

**Review by amit-yb — APPROVED** [review]
Approved (with a very minute comment).

**Pull request merged**